libraries/chain/include/eosio/chain/hotstuff/hotstuff.hpp

```diff
@@ -190,8 +183,23 @@ namespace eosio::chain {
       state_t              _state { state_t::unrestricted };
       size_t               _num_finalizers {0};
       size_t               _quorum {0};
+      std::unique_ptr<std::mutex> _mtx;  // protect both _strong_votes and _weak_votes
       votes_t              _weak_votes;
       votes_t              _strong_votes;
+
+   private:
+#warning TODO move members above to private after unification. Currently they are used by qc_chain.cpp directly
```

Comment:
Can you make qc_chain a friend. Very hard to determine if this is thread-safe as-is.